File: Leetcode/Easy/27.RemoveElement.py
# Given an integer array nums and an integer val, 
# remove all occurrences of val in nums in-place. 
# The relative order of the elements may be changed.

def removeElement(array, x):
    if len(array) == 1 and array[0] == x:
        return 0

    index = None

    for i in range(len(array)):
        if array[i] == x and index == None:
            index = i
        elif array[i] != x and index != None:
            array[index] = array[i]
            index += 1

    return array

print(removeElement([2,2,2,2,3,0,4,2], 2))
print(removeElement([1,2], 1))
print(removeElement([3,2,2,3], 3))

File: NeetCode/Swift/Arrays/Easy/27.RemoveElement.swift
//Given an integer array nums and an integer val,
//remove all occurrences of val in nums in-place. 
//The relative order of the elements may be changed.


func removeElement(_ nums: inout [Int], _ val: Int) -> Int {
    var index = nums.count - 1
    var count = 0

    for i in stride(from: nums.count - 1, to: -1, by: -1) {
        if nums[i] == val {
            nums.swapAt(i, index)
            index -= 1
            count += 1
        }
    }

    return nums.count - count
}

print(removeElement([3,2,2,3], 3))
print(removeElement([0,1,2,2,3,0,4,2], 2))